### PR TITLE
Threads: Wait Counter

### PR DESCRIPTION
**Pull request by denisdimitrov**

### CODE REVIEW

File: src/main/java/com/clouway/Counter.java [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+                    z.printStackTrace();
+                }
+            }
+        }
+    }
+
+    public static void main(String[] args) {
+        Object lock = new Object();
+        Thread thread = new Thread(new Counter(5, lock));
+        Thread thread1 = new Thread(new Counter(5, lock));
```
Comment (EmilGeorgiev) [inline src/main/java/com/clouway/Counter.java]: Now if we pass different number of seconds for each thread program will block i never down. I think when one thread finish with its work should stop another. Simply Modify second task

Comment (denisdimitrov) [inline src/main/java/com/clouway/Counter.java]: Done.

File: src/main/java/com/clouway/Counter.java [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+/**
+ * Created by clouway on 26.09.16.
+ *
+ * @author Denis Dimitrov <denis.k.dimitrov@gmail.com>
+ */
+public class Counter implements Runnable {
+    private int seconds;
+    private int counter;
+    private AtomicBoolean flag;
+    private final Object lock;
```
Comment (EmilGeorgiev) [inline src/main/java/com/clouway/Counter.java]: you can use ```AtomicBoolean``` instead ```Object``` for synchronized  block

**Comment by EmilGeorgiev** [comment]
LGTM after fix comment

**Pull request merged**